include/seastar/core/abort_source.hh

```diff
@@ -136,20 +147,17 @@ public:
     /// Requests that the target operation be aborted. Current subscriptions
     /// are invoked inline with this call, and no new ones can be registered.
     /// Must be called exactly once, otherwise the program will be aborted.
-    void request_abort(std::optional<std::exception_ptr> ex = std::nullopt) noexcept {
-        assert(_subscriptions);
-        _ex = ex.value_or(get_default_exception());
-        auto subs = std::exchange(_subscriptions, std::nullopt);
-        while (!subs->empty()) {
-            subscription& s = subs->front();
-            s.unlink();
-            s.on_abort(ex);
-        }
+    void request_abort() noexcept {
+        do_request_abort(std::nullopt);
+    }
+
+    void request_abort_ex(std::exception_ptr ex) noexcept {
+        do_request_abort(std::make_optional(std::move(ex)));
```

Comment:
Ugh, that is sad.

Reply:
Please add doxygen comments to the new function.

Reply:
done in v2